File: sdks/python/read_dir/template/res.py
from typing import Optional, List, Dict, Union

from api.base_model import CyanBaseModel
from api.core.cyan_res import CyanRes
from api.core.question_res import QuestionRes


class TemplateValidRes(CyanBaseModel):
    valid: Optional[str]


class TemplateFinalRes(CyanBaseModel):
    cyan: CyanRes
    type: str = "final"


class TemplateQnARes(CyanBaseModel):
    deterministic_state: List[Dict[str, str]]
    question: QuestionRes
    type: str = "questionnaire"


# Union type for TemplateRes
TemplateRes = Union[TemplateQnARes, TemplateFinalRes]

File: sdks/node/src/api/template/res.ts
import type { CyanRes } from '../core/cyan_res.js';
import type { QuestionRes } from '../core/question_res.js';

interface TemplateValidRes {
  valid?: string | null; // Use optional property for nullability
}

interface TemplateFinalRes {
  cyan: CyanRes;
  type: 'final';
}

interface TemplateQnARes {
  deterministicState: Record<string, string>;
  question: QuestionRes;
  type: 'questionnaire';
}

type TemplateRes = TemplateQnARes | TemplateFinalRes;

// export all
export type { TemplateRes, TemplateQnARes, TemplateFinalRes, TemplateValidRes };
